### DISCUSSION TITLE
easier composer install local

### DISCUSSION THREAD

**Discussion by dillingham** [composer/composer#10201] [Q&A / Support] [answered]
with npm, you can `git clone repo && npm i ./repo` and be done

would be nice if you `composer install ./some-path` it automatically adds it to a `repositories` and `require` and dumps autoload

**Comment by Seldaek** [comment]
Yeah I guess the way this would work in composer is `composer config repo.foo vcs https://github.com/... && composer require new/pkg --prefer-source`, then you have it cloned in vendor/new/pkg

Or alternatively `git clone repo && composer config repo.foo path ./repo && composer require new/pkg` would be more like what you are asking I suppose.

It's not quite the same, but it's also not that much IMO. I'm not sure how often you need to do this sort of thing?

**Reply by dillingham** [reply]
Thanks! I mixed your help into a bash helper :)

**Comment by dillingham** [comment, marked as answer]
Here is a little bash helper I made

```
composer_local ./path/to/package
```
```bash
composer_local() {
    package=$(cat $1/composer.json | jq -r ".name")
    composer config repo.$package path $1
    composer require $package @dev
}
```
